### ISSUE TITLE
sapper upgrade

### ISSUE THREAD

**Issue by Rich-Harris** [sveltejs/sapper#92]
#91 proposes changing the required project structure so that `2xx.html` becomes `page.html`, and `4xx.html` and `5xx.html` are moved into the `routes` folder.

#86 removed `%sapper.main%` in favour of `%sapper.scripts%`.

These probably aren't the last times we'll ever change our minds about how something should be organised. Changes like these can be a pain for people upgrading, and in a worst case scenario can discourage them from doing so.

While I'm not a fan of the walled garden/black box/'toolkit' approach, for reasons I've [articulated](https://github.com/sveltejs/sapper/issues/68) [elsewhere](https://twitter.com/Rich_Harris/status/953005486969966595), it is nice to provide as seamless an upgrade path as possible.

One possible approach would be a `sapper upgrade` command that detects any deprecated constructs and fixes them with codemods etc:

```bash
$ yarn start

templates/2xx.html
<script src='%sapper.main%'> is unsupported — use %sapper.scripts% (without the <script> tag) instead
https://github.com/sveltejs/sapper/issues/86
   34 |        initialise the router -->
 > 35 |   <script src='%sapper.main%'></script>
      |                ^

Run `sapper upgrade` to fix this issue automatically.
```

**Comment by Rich-Harris** [comment]
Since `degit` is just based on git repos, you could either pull the most recent master branch of sveltejs/sapper-template, or manually look at the commits to see what's changed. But `sapper upgrade` will be in the next release (I just merged #120)

**Comment by Rich-Harris** [comment]
Closing, as this is implemented (though there's plenty more that could be done in terms of specific upgrade tasks)
